### DISCUSSION TITLE
Debugging with gds and gdb with arguments

### DISCUSSION THREAD

**Discussion by Emil808** [nasa/fprime#1453] [Q&A]
In extension to #1232 and #918. 

I have a deployment that I want to debug that is hitting an assert way to fast to do the method in #1232. 

When trying to do the method in #918, How do I format the inputs for the deployment to connect to the gds?

**Comment by Emil808** [comment]
With this stack overflow thread: https://stackoverflow.com/questions/14494957/how-do-i-pass-a-command-line-argument-while-starting-up-gdb-in-linux

Extending the method in #918, 

```
fprime-util generate --ut <platform or native>
cd fprime-build-automatic-<platform>-ut
make <deployment>_exe
gdb --args ./bin/*/<deployment> -a <IP of gds> -p <port of gds> 
```

**Reply by LeStarch** [reply]
That looks right to me.  As a note the default IP is: `127.0.0.1` and the default port is `50000` so:

```
fprime-gds -n # Run gds without starting app
gdb --args ./bin/*/<deployment> -a 127.0.0.1 -p 50000
```
